pkg/controller/cassandracluster/generator_test.go

```diff
@@ -330,6 +339,30 @@ func TestInitContainerConfigFileData(t *testing.T) {
 	assert.Equal(configFileData.String(), initEnvVar[0].Value)
 }
 
+func TestInitContainerServerVersionDetect(t *testing.T) {
+	dcName := "dc1"
+	rackName := "rack1"
+	dcRackName := fmt.Sprintf("%s-%s", dcName, rackName)
+
+	_, cc := helperInitCluster(t, "cassandracluster-1DC.yaml")
+	// Cassandra image contains tag with cassandra version
+	imageTag := "1.0.0"
+	cc.Spec.CassandraImage = fmt.Sprintf("artifactory-name/repo/cassandra:%s", imageTag)
+	cassieResources := cc.Spec.Resources
+
+	assert := assert.New(t)
+	initEnvVar := initContainerEnvVar(cc, &cc.Status, cassieResources, dcRackName)
+	prodVer := GetEnvVarByName(initEnvVar, "PRODUCT_VERSION")
+	assert.Equal(imageTag, prodVer.Value)
+
+	// Cassandra image contains custom port for artifactory and cassandra version
+	imageTag = "1.0.1"
+	cc.Spec.CassandraImage = fmt.Sprintf("artifactory-name:44301/repo/cassandra:%s", imageTag)
+	initEnvVar = initContainerEnvVar(cc, &cc.Status, cassieResources, dcRackName)
+	prodVer = GetEnvVarByName(initEnvVar, "PRODUCT_VERSION")
+	assert.Equal(imageTag, prodVer.Value)
+}
```

Comment:
@ibumarskov please refactor your code as a loop instead. You can keep the same imageTag and just try both format with the same code.

Reply:
Done